numpy/linalg/linalg.py

```diff
@@ -1583,15 +1586,16 @@ def pinv(a, rcond=1e-15 ):
     _assertNoEmpty2d(a)
     a = a.conjugate()
     u, s, vt = svd(a, 0)
-    m = u.shape[0]
-    n = vt.shape[1]
-    cutoff = rcond*maximum.reduce(s)
-    for i in range(min(n, m)):
-        if s[i] > cutoff:
-            s[i] = 1./s[i]
-        else:
-            s[i] = 0.;
-    res = dot(transpose(vt), multiply(s[:, newaxis], transpose(u)))
+    m = u.shape[-2]
+    n = vt.shape[-1]
+    cutoff = rcond * maximum.reduce(s, axis=-1, keepdims=True)
+    mask = s > cutoff
+    s[mask] = 1. / s[mask]
+    s[~mask] = 0
```

Comment:
The arguments are documented on the [ufuncs](https://docs.scipy.org/doc/numpy/reference/ufuncs.html#optional-keyword-arguments) page. I'd argue that we should remove the documentation of the `out` argument from all the ufuncs, and just include a reference back to the full documentation.